### PR TITLE
composer: add support for logrus syslog hook

### PR DESCRIPTION
**Pull request by diaasami**
Which will be used on crc in the log forwarding setup
NeededBy: COMPOSER-1285

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: cmd/osbuild-composer/config.go [hunk 1]
```diff
@@ -11,11 +11,12 @@ import (
...
 type ComposerConfigFile struct {
-	Koji      KojiAPIConfig   `toml:"koji"`
-	Worker    WorkerAPIConfig `toml:"worker"`
-	WeldrAPI  WeldrAPIConfig  `toml:"weldr_api"`
-	LogLevel  string          `toml:"log_level"`
-	LogFormat string          `toml:"log_format"`
+	Koji         KojiAPIConfig   `toml:"koji"`
+	Worker       WorkerAPIConfig `toml:"worker"`
+	WeldrAPI     WeldrAPIConfig  `toml:"weldr_api"`
+	SyslogServer string          `toml:"syslog_server" env:"SYSLOGSRVR"`
```
Comment (ondrejbudai) [inline cmd/osbuild-composer/config.go:17]: I would probably go with `SYSLOG_SERVER`, I don't think we need to save bytes,

File: cmd/osbuild-composer/main.go [hunk 2]
```diff
@@ -69,6 +70,15 @@ func main() {
 		logrus.Fatal("CACHE_DIRECTORY is not set. Is the service file missing CacheDirectory=?")
 	}
 
+	if len(config.SyslogServer) > 0 {
+		hook, err := syslog.NewSyslogHook("tcp", config.SyslogServer, 0, "osbuild-composer")
+		if err != nil {
+			panic(err)
```
Comment (ondrejbudai) [inline cmd/osbuild-composer/main.go:76]: `logrus.Fatal` for the sake of consistency?

**Review by ondrejbudai — COMMENTED** [review]
Can you please rebase this to get rid of the merge commit? Otherwise, this looks nice, thank you. Btw, I was thinking of writing a small test for this but I don't see an obvious way unfortunately.

File: cmd/osbuild-composer/config.go (hunk 1 above)
Comment (diaasami) [reply cmd/osbuild-composer/config.go:17]: I agree `SYSLOG_SERVER` is better, I made it this way for consistency with other env vars like `PGUSER` and `PGPASSWORD`

Comment (Gundersanne) [reply cmd/osbuild-composer/config.go:17]: PG* etc. are standard postgres variables https://www.postgresql.org/docs/current/libpq-envars.html, so I just used the colloquial ones there.

**Review by ondrejbudai — APPROVED** [review]

**Review by Gundersanne — APPROVED** [review]

**Pull request merged**